src/dataflow/src/source/postgres.rs

```diff
@@ -138,47 +159,25 @@ impl PostgresSourceReader {
             .parse()
             .or_else(|_| Err(anyhow!("invalid lsn")))?;
 
-        let copy_query = format!(
-            r#"COPY {} TO STDOUT WITH (FORMAT binary);"#,
-            self.connector.ast_table,
-        );
-
-        let stream = client.copy_out_simple(&copy_query).await?;
-
-        let scalar_types = table_info
-            .schema
-            .iter()
-            .map(|c| c.scalar_type.clone())
-            .collect_vec();
-
-        let rows = BinaryCopyOutStream::new(stream, &scalar_types);
-        tokio::pin!(rows);
-
         let snapshot_tx = timestamper.start_tx().await;
-        while let Some(row) = rows.next().await.transpose()? {
-            let mut mz_row = Row::default();
-            for (i, ty) in table_info.schema.iter().enumerate() {
-                let datum: Datum = match ty.scalar_type {
-                    PgType::BOOL => row.get::<Option<bool>>(i).into(),
-                    PgType::INT4 => row.get::<Option<i32>>(i).into(),
-                    PgType::INT8 => row.get::<Option<i64>>(i).into(),
-                    PgType::FLOAT4 => row.get::<Option<f32>>(i).into(),
-                    PgType::FLOAT8 => row.get::<Option<f64>>(i).into(),
-                    PgType::DATE => row.get::<Option<NaiveDate>>(i).into(),
-                    PgType::TIME => row.get::<Option<NaiveTime>>(i).into(),
-                    PgType::TIMESTAMP => row.get::<Option<NaiveDateTime>>(i).into(),
-                    PgType::TEXT => row.get::<Option<&str>>(i).into(),
-                    PgType::UUID => row.get::<Option<Uuid>>(i).into(),
-                    ref other => bail!("Unsupported data type {:?}", other),
-                };
 
-                if !ty.nullable && datum.is_null() {
-                    bail!("Got null value in non nullable column");
+        for (rel_id, schema, table) in publication_tables {
+            let data = client
```

Comment:
Oh, so different! Do you know how the performance of `SELECT * FROM x` compares to `COPY {} TO STDOUT`? For future versions, do you think it might be worthwhile to have multiple `COPY {} to [table file]` calls, instead?

Reply:
The difference is only in the network format which has a bit more overhead than just `COPY`. I switched to a query because we can no-longer use the `BINARY` COPY format and rust-postgres doesn't provide decoders for the textual one. Nevertheless, this happens once on initialization so it can be optimized later. I'll leave a comment to that effect